src/goto-programs/remove_exceptions.cpp

```diff
@@ -508,13 +513,40 @@ void remove_exceptions(
   const namespacet ns(symbol_table);
   std::map<irep_idt, std::set<irep_idt>> exceptions_map;
   uncaught_exceptions(goto_functions, ns, exceptions_map);
+  auto function_may_throw = [&exceptions_map](const irep_idt &id) {
+    return !exceptions_map[id].empty();
+  };
   remove_exceptionst remove_exceptions(
     symbol_table,
-    exceptions_map,
+    function_may_throw,
     type == remove_exceptions_typest::REMOVE_ADDED_INSTANCEOF);
   remove_exceptions(goto_functions);
 }
 
+/// removes throws/CATCH-POP/CATCH-PUSH from a single GOTO program, replacing
+/// them with explicit exception propagation.
+/// Note this is somewhat less accurate than the whole-goto-model version,
+/// because we can't inspect other functions to determine whether they throw
+/// or not, and therefore must assume they do and always introduce post-call
+/// exception dispatch.
```

Comment:
Consider explaining why we would use this version vs. the more precise version above?